### DISCUSSION TITLE
Show Overlay frame on the live view

### DISCUSSION THREAD

**Discussion by Striker62** [photobooth-app/photobooth-app#352] [Show and tell]
Hi,

if you like to have your overlay frame also on the live view.
Here is the code for the private .css

#preview-stream::after {
    content: "";
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background-image: url('/userdata/frames/frame.png'); /* Dein PNG als Overlay */
    background-size: cover; /* Deckt den gesamten Bereich ab */
    background-position: center;
    pointer-events: none; /* Erlaubt weiterhin die Interaktion mit dem Stream */
    z-index: 1; /* Stellt sicher, dass das Overlay über dem Stream liegt */
    opacity: 0.9; /* Optional: Transparenz für das Overlay */
/*    transform: scaleX(-1); /* Horizontal spiegeln */*/

I use a Pi Cam and a 7" touchscreen

here my frame 

![frame](https://github.com/user-attachments/assets/b1bc2960-ebd7-4907-9d69-820d8d1b215f)

**Comment by mgineer85** [comment]
Thank you, I added this to the docs: https://photobooth-app.org/reference/customizetheme/#add-a-frame-to-the-frontpage

**Comment by rektosaure** [comment]
Hi,

Many thanks for this.

I also would like to dynamically apply a theme (frame) to the live preview based on the theme selected by the user through icons in the interface.
I would like to know if it’s possible to change the appearance of the livestream depending on the selected theme, using only CSS with private.css.
So far, I have managed to add a static frame to the livestream (as described in the conversation above), but I am unable to switch this frame dynamically based on the selected theme.

Thanks in advance for your help!

**Reply by Striker62** [reply]
Hi, i Named my Frame "Frame.png" in the userdata/frame folder.
In my private.css i use this, so there is in Need to change the css. You only have to manage to rename the Frame you whant to use in Frame.png. This can be done By a script That you can call in a Share Action as for the printing.
The question is how to Show the User what Frame je can select and the script shell rename to frame.png.

I Hope this help you to find your Solution.

**Reply by Striker62** [reply]
To make sure that the actual Frame is shown and not one in cache you can add the following to the frame url: ?v=1.0

background-image: url('/userdata/frames/frame.png?v=1.0');

**Reply by rektosaure** [reply]
I understand the idea, very interesting.
The question is indeed: how does the script detect that the user has pressed a specific button corresponding to a specific theme?
Maybe @mgineer85 can help?

**Reply by Striker62** [reply]
Hi, I tried to show it here "Frame selection Howto #356"  please have a look

**Reply by rektosaure** [reply]
I saw your post, thanks for sharing.
What I’m trying to achieve is a “live” theme change. On the livestream, the user can choose between 2 image icons for example, each with a different theme.
The frame changes when the user clicks on one of the icons and the countdown starts. This way, the livestream is dynamic and not limited to a single theme like in the solution you proposed.

**Reply by Striker62** [reply]
ok understood you would like to have 2 take a picture buttons . That can only be done by changing the vue code I think. 
Probably if my 2 buttons can placed on the live view wie can add an api cal after the file copy string to take a picture .
By the way a add character "b" for the birthday frame and "w" forn the wedding . so if you provide to bluetooth button the frame can be changed dirct on the live view. but this is not a solution for yur one click requirement.

**Reply by mgineer85** [reply]
Hey, currently there is no way I am aware of to dynamically change the frame by css only.
There is no way to select different actions during preview only currently.   To have it I could think about some kind of selection button next to the action (a toggle for example). But this is some additional element on the frontpage and needs to be understood how to use. 
 On contrary, a user usually just don't want to bother long time and just get the photo done.

**Reply by rektosaure** [reply]
Hi @mgineer85,

Thanks for your response. Just to clarify, as I’m not sure we’re on the same page yet.
My request is not about adding new buttons where the user would choose a theme before taking the photo. Instead, the icons corresponding to different themes are already predefined in the configuration.

What I would like to achieve is that, when the user selects one of these predefined icons, the corresponding theme is automatically applied to the live video feed during the countdown before the photo is taken. So, the theme dynamically overlays the live stream from the moment the user clicks the icon, without requiring any additional selection steps or buttons.

I hope this helps to clarify my request. Thanks again for your help!

**Reply by mgineer85** [reply]
Hi Right, understood.
I was thinking about how to handle the preview even before a button is pushed.
The user would see the theme only from the moment of pressed button. This could work ok for long countdown but not so well to setup in front of the booth.

I think we could add a dynamic class after triggering and so handle the frame after button is pushed more convenient.

**Reply by Striker62** [reply]
Hi,
from my point of view every thing is there if you watch my video. Only the live view refresh is missing.
If you have any idea how a refresh can be triggert by api call or an char that can be send via keyboard it should be very close to the requierement

**Comment by rektosaure** [comment]
Maybe something like that in private.css could work?

```
/* Theme 1 */
.action-button-0:checked` ~ #preview-stream::after {
    background-image: url('/userdata/themes/theme1.png');
}

/* Theme 2 */
.action-button-1:checked ~ #preview-stream::after {
    background-image: url('/userdata/themes/theme2.png');
}
```

Couldn’t make it work as my CSS knowledge is really limited.

**Reply by mgineer85** [reply]
That's kind of cool idea, but the toggle button is missing to make it :checked .

**Comment by Striker62** [comment]
![grafik](https://github.com/user-attachments/assets/27e7e58b-f2a2-4992-9d63-2d6bd9c82af0)
In the config for the share action there is a switch that is miss leading ? 
It say on Main screen but it is on the media section detail screen if i'm not wrong .
can you check

**Reply by mgineer85** [reply]
Oh yeah - it's the same object that is used for UI buttons for actions and share. I will change the text.

**Reply by Striker62** [reply]
I hoped you will change it that it works as descriped :-)

**Comment by PixelAlarm** [comment]
Is there some way to show the frame overlay in idle (no button clicked) and hide it when the countdown starts?
That way it would be really easy to theme the welcome-screen. Best regards

**Reply by mgineer85** [reply]
You mean like this https://photobooth-app.org/reference/customizetheme/#add-a-frame-to-the-frontpage ?

**Reply by PixelAlarm** [reply]
Exactly, but since I‘d like to have a fullscreen overlay as png it shouldn‘t be displayed while countdown / capture.
